components/constellation/constellation.rs

```diff
@@ -2470,7 +2464,7 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
                 .filter_map(|maybe_pipeline| maybe_pipeline)
                 .collect::<Vec<_>>();
 
-            pipelines_to_evict.extend(session_history.future.iter()
+            pipelines_to_evict.extend(session_history.future.iter().rev()
```

Comment:
Really? The session history future is in increasing chronological order isn't it? We should be evicting ones far off in the future, not the near future.

Reply:
Yes we want to reverse here because this will put the diffs in the near future first, which we then skip by `history_length`.